src/dbos-executor.ts

```diff
@@ -423,8 +431,23 @@ export class DBOSExecutor {
 
     const runWorkflow = async () => {
       let result: R;
+      // Track concurrent executions of the WF
+      if (!this.workflowStats.has(wf.name)) {
+        this.workflowStats.set(wf.name, new WorkflowStats());
+      }
+      const wfs = this.workflowStats.get(wf.name)!;
+      wfs.currentlyExecuting++;
+      wfs.nStarted++;
+
       // Execute the workflow.
       try {
+        // If this is a scheduled workflow, do the pre-workflow recordkeeping
+        if (wInfo.config.registrationObject && (wInfo.config.registrationObject as SchedulerRegistrationConfig).schedulerConfig) {
+          const inargs = args as unknown as ScheduledArgsAsSerialized;
+          await this.systemDatabase.scheduledWorkflowStarted(wf.name, new Date(inargs[0]).getTime());
+          const nRunningGlobally = await this.systemDatabase.getOutstandingScheduledWorkflows(wf.name);
+          args = [inargs[0], inargs[1], nRunningGlobally, wfs.currentlyExecuting] as unknown as T;
```

Comment:
An issue with updating input args here is that, if you restart a workflow, it changes the input args after we return from recorded inputs. It could be an issue if a workflow with the same UUID gets different inputs across runs and it complicates our programming model (workflows with the same UUID should get the same inputs across restarts).

Reply:
In this case we do not want the same values across restarts.  But I could move those values from the parameter list to the context or something like that to make it clearer that they don't behave completely like parameters?

Reply:
I've decided there is a fundamental problem with how it is, and will go back to thinking about this.